spec/registry/unicode.md

```diff
@@ -0,0 +1,63 @@
+# MessageFormat 2.0 Unicode Registry
+
+The `u:` namespace is reserved for use by the Unicode Consortium.
+
+## Options
+
+This section describes common options which each implementation SHOULD support
+for all _functions_ and _markup_.
+
+### `u:id`
+
+A string value that is included as an `id` or other suitable value
+in the formatted parts for the _placeholder_,
+or any other structured formatted results.
+
+Ignored when formatting a message to a string.
+
+Accepts string values, or values which can be stringified without error.
+For other values, a _Bad Option_ error is emitted
+and the `u:id` option is ignored.
+
+### `u:locale`
+
+A comma-delimited list of BCP 47 language tags,
+or an implementation-defined list of such tags.
+
+Replaces the _locale_ defined in the _function context_ for this _expression_.
+The value is ignored when set on _markup_.
+
+During processing, the `u:locale` option
+is always removed from the resolved mapping of _options_.
+
+Values matching the following ABNF are always accepted:
+```abnf
+u-locale-option = langtag *([s] "," [s] langtag)
+```
+using `langtag` as defined in [RFC 5646](https://www.rfc-editor.org/rfc/rfc5646).
+
+Implementations MAY support additional language tags,
+such as private-use or grandfathered tags,
+or tags using `_` instead of `-` as a separator.
+When the value of `u:locale` is set by a _variable_,
+implementations MAY support non-string values otherwise representing locales.
```

Comment:
Some problems here. Private-use and grandfathered tags are already covered by both the `langtag` production and BCP47 in general.

I think you're trying to do several things here and we should split them apart:

```suggestion
Implementations MAY process the list of locale identifiers,
including interpretation of language tags or canonicalization of values 
(such as mapping grandfathered tags to modern representations).

Implementations MAY convert or map the list to a prioritized list of
implementation-specific values.
For example, a JavaScript implementation might convert it into an `Intl.Locale` 
while a Java implementation might convert the list into an array of `java.util.Locale` objects.
Implementations MAY reject unrecognized or unsupported values.

Implementation MAY support proprietary or implementation-specific
locale identifiers.
```

Reply:
I updated the previous paragraph to note that as we're specifically referring to the `langtag` rule and not the `Language-Tag` rule, we're not requiring support for private-use or grandfathered tags.

The intended point of this passage is to make sure that values not matching `u-locale-option` may still be accepted, and explicitly calling out the extensions that are required for either BDP47 or ULI support.

Reply:
I don't think this is the right move. We don't need to get into the locale identifier game. I think referencing `langtag` instead of `Language-Tag` is a backwards step.

Do you not agree with the various changes I suggested?

Reply:
I agree that we should not get into the locale identifier game. So much as we did with `:datetime` literals, I think we should identify a minimal definition that must be supported, and allow for implementations to go beyond that. And I think the `langtag` rule offers that minimum, i.e. something that can be supported by both a BCP47 and a ULI implementation, where both of those have features unsupported by the other.

Reply:
Replaced `langtag with `unicode_bcp47_locale_id`.